server/src/test/java/org/opensearch/snapshots/SnapshotResiliencyTests.java

```diff
@@ -686,6 +686,55 @@ public void testConcurrentSnapshotCreateAndDeleteOther() {
         }
     }
 
+    public void testTransportGetSnapshotsAction() {
```

Comment:
@indrajohn7  now that we have separated the flow for currentSnapshotOnly, please add a coverage for that same with 1 current snapshot in progress.

Reply:
You can cover currentSnapshot as below using ClusterStateListener which can check if snapshot is in progress

```
       CountDownLatch latch = new CountDownLatch(1);
        clusterManagerNode.clusterService.addListener(new ClusterStateListener() {
            @Override
            public void clusterChanged(ClusterChangedEvent event) {
                if (event.state().custom(SnapshotsInProgress.TYPE) != null) {
                    // assert here for in progress snapshot and call your transport get snapshot action here
                    SnapshotsInProgress snapshotsInProgress = event.state().custom(SnapshotsInProgress.TYPE);
                    assertTrue(snapshotsInProgress.entries().stream().anyMatch(entry -> entry.state().completed() == false));
                    latch.countDown();
                }
            }
        });

       continueOrDie(createRepoAndIndex(repoName, index, shards), createIndexResponse -> {
            final Runnable afterIndexing = () -> client().admin()
                .cluster()
                .prepareCreateSnapshot(repoName, snapshotName)
                .setWaitForCompletion(false)     // set this to false so that it is unblocked.
                .execute(createSnapshotResponseListener);
```

Reply:
Ack.